### DISCUSSION TITLE
*.bin file to flash orig. wifi Dongle

### DISCUSSION THREAD

**Discussion by tomm1st** [syssi/esphome-soyosource-gtn-virtual-meter#78] [General]
Hello, is there an easy to use *.bin file to flash on the original Wifi Dongle, to get it running "clound free"? Yust want to have an webinterface, no home assit, oder ESP home in my system...
Thx.

**Comment by syssi** [comment]
Would you be willing to learn how to use/build ESPHome setups (without Home Assistant) or are you looking for a "flash & forget" solution? Do you plan to use the virtual limiter or do you want to monitor your inverter only?

**Reply by syssi** [reply]
I'm asking because installing `esphome` as command line tool and use it as "compiler & flash" tool is pretty easy:

![Screen recording](https://raw.githubusercontent.com/syssi/esphome-seplos-bms/main/install.gif)

**Reply by syssi** [reply]
~I've prepared a standalone (web-only) version.~

~It can be flashed easily via: https://web.esphome.io/~

~Please don't forget to set your WiFi credentials via web.esphome.io too ("..." -> Configure Wi-Fi).~

~The web interface will look like this:~

![gtn-display-web](https://user-images.githubusercontent.com/2735933/192092161-bae39dbe-2910-4a6c-95b5-64327e66d480.png)

Hmm.. it doesn't work. ;-) Using `improv` to provision the device isn't the solution.

**Reply by syssi** [reply]
I've prepared another version providing a captive portal (SSID: "wifi-setup", Password: "soyosource") for the initial WiFi setup.

Please flash this firmware image via web.esphome.io:

[soyosource-wifi-dongle-web-only-1.1.0.zip](https://github.com/syssi/esphome-soyosource-gtn-virtual-meter/files/9638469/soyosource-wifi-dongle-web-only-1.1.0.zip)

Let the dongle boot and look for a WiFi network called "wifi-setup". Have fun!

**Reply by tomm1st** [reply]
Thats great! Thank you so much! i will try to flash this evening and let you know, if it works.
Answer to your first question: I have installes ESPhome yesterday night, and maybe i would be able to flash now also on that way, but before i can flash anything, i need to get a connection to the original dongle what i don`t get running, but it works on an wemos D1.... so more ore less onyl some issues with the right connection, what is maybe somehow due to the point, that i use ftdi chip of the Wemos D1 board, have the ESP there permanently reset, and connect RX/TX to the dongle.... that should work in my eyes in principle, but maybe i have an failure in this idea and need to wait for my ftdi usb cable i ordered.
Best regards Thomas

**Reply by syssi** [reply]
Sounds good. Just keep me updated! Let know if you run out of ideas. It's super important to pull down the RST of the second microcontroller on the WiFi dongle because it's wired to GPIO1/GPIO3 of the ESP8266 and will disturb the flash operation.

**Reply by tomm1st** [reply]
may my next try is to solder all thouse things, may my temporry contacts are not tooo reliable and that is the reason, but i have to try this timorrow after work.
Best regards

**Reply by tomm1st** [reply]
So, i got it running! with your soyosource-wifi-dongle-web-only-1.1.0.zip *.bin File
So thats great because i can now continue with my Project.
But I´m also a litte bit disapointed, that i don`t get it running with my own complied file.... when i compile and flash it with ESPhome via terminal command, it doesn`t setup a first wifi....
Now not a real problem anymore, but i would be interesed in finding my failure.... to learn for future projects...
do you have an idea?
best regards

**Reply by syssi** [reply]
Did you set the WiFi credentials via the configuration YAML or did you add the `captive_portal` component + access point mode?

**Reply by tomm1st** [reply]
I set the wifi credentials in the "secret,yaml" as in the *.gif you added me in the 2nd post
What you are writing with the second MC that i need to pull down rst was maybe the main issue, the way i got it running was, to flash your file on an esp8266 board, and then exchange the chip with the one on the soyo dongle...

**Comment by odroegehorn** [comment]
Hi all,
is it also possible to connect the RS485 adapter to the stock-wifi-dongle with the new firmware? Are there free GPIOs for this?

If NOT:
How to connect a WeMos to the USB-Port? Do I have to use USB-TTL converter? I guess I can't directly connect the USB RX/TX to the GPIOs of an 8266?

Thanks,
kind regards,
Olaf

**Reply by syssi** [reply]
Yes it should be possible. You should pick some free GPIOs next to the ESP case (without traces on the PCB).

**Reply by odroegehorn** [reply]
Got it connected and soldered. After a recompile, also the limiter works.

But now: How to send via MQTT values to the limiter? Is it documented somewhere?
Thanks,
best,
Olaf

**Reply by odroegehorn** [reply]
Sorry, another question:
How can I reduce the interval, in which MQTT messages are sent? Or even the messages on the web-page, that are shown?
It's talking all the time.
Thanks.

**Reply by syssi** [reply]
You should find a closed issue with some instructions: https://github.com/syssi/esphome-soyosource-gtn-virtual-meter/issues/71

**Reply by odroegehorn** [reply]
Thanks, but it's only shown, that the powermeter can be replaced to subscribe itself to another device.
That's coo, as long as that device publishes the things.
My Powermeter is an Eastron SDM630 an I get the information (in FHEM) vie ModBus.
So is there a way to send the deman-power from my server (FHEM with MQTT) directly to the virtual powermeter?

The goal is, that your software in the Wifi-Stick is NOT subsribing itself to something, but wants to recieve a new value via an MQTT command.
Sorry for that, my setup is slightly different.

BTW: the RS485 on the back of the Wifi-Dongle still fit's into it's original housing. Therefore no extra-platstics, just a cable from the dongle to the RS485 port. 
Thanks so far

**Reply by syssi** [reply]
You could reduce the update interval of the display component:

```
soyosource_display:
  uart_id: uart0
  update_interval: 30s
```

Please try to understand the concept of MQTT. ;-) It looks like you are missing something here. The virtual meter (ESPHome node) must subscribe to topic A and your FHEM instance has to publish the power demand periodically to topic A. This is how MQTT works.

**Reply by syssi** [reply]
> BTW: the RS485 on the back of the Wifi-Dongle still fit's into it's original housing. Therefore no extra-platstics, just a cable from the dongle to the RS485 port.
Thanks so far

Could you provide a photo and the configuration (used GPIOs) of your setup? I would like to improve the documentation and suggest which GPIOs can be used.

**Reply by odroegehorn** [reply]
ok, great.
With MQTT it's fine. Thought there might be a predefined topic to publish towards. OK, I will define one.

Photo: I can do this. Where to post? here?
I used GPIO5 (as it is free) and GPIO14 (as it was presoldered), easiest way to solder for me. Originally I planned for GPIO5 & 16, but the 16 PIN didn't got strong soldering for the cable, therefore I switched to GPIO14.

Will send a photo tomorrow.

**Reply by syssi** [reply]
No there is no predefined topic. In general only a ESPHome sensor entity is required to drive/feed the component. You could replace the MQTT based sensor entity by any other sensor (f.e. a template sensor updated via the HTTP/REST API). You don't need to stick to MQTT but it's the recommended solution if you don't use Home Assistant (+ the native API).
